File: 55. Jump Game/canJump.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
const canJump = (nums) => {
  let goal = nums.length - 1;

  for (let i = nums.length - 1; i >= 0; i-- ) {
    if (i + nums[i] >= goal) {
      goal = i;
    }
  }

  return goal === 0;
};


const nums = [2, 3, 1, 1, 4];
console.log('canJump: ', canJump(nums));

File: 55. Jump Game/canJump.py
class Solution:
  def canJump(self, nums: list[int]) -> bool:
    goal = len(nums) - 1

    for i in range(len(nums) - 1, -1, -1):
      if i + nums[i] >= goal:
        goal = i

    return goal == 0


nums = [2, 3, 1, 1, 4]
solution = Solution()
print('canJump: ', solution.canJump(nums))
